### ISSUE TITLE
game crash when loading savegames during the game 

### ISSUE THREAD

**Issue by RBRiddick** [We-the-People-civ4col-mod/Mod#625]
Hello. First of all thank you 4 your great job.

Game crash when I load a savegame while I'm playing.
As example, I plan to do a critical move. I save the game and try it. When I load the savegame, to test other endings, the game is shutting down. Restarting the game and loading the SAME save, all works.
I'm playing under linux with wine, but same happens under windows I have on an other partition.
The wine error message is approximately this: The program Colonization.exe encountered an error and must be closed. 
Backtrace: 
[backtrace.txt](https://github.com/We-the-People-civ4col-mod/Mod/files/9847461/backtrace.txt)

I've tried to change same graphic settings and remove fullscreen mode, but nothing change.
Here is a saved game:
[Penn d.C. - 1503-Maggio.ColonizationSave.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/9847416/Penn.d.C.-.1503-Maggio.ColonizationSave.zip)

My configurations:
linux mint 18.3 Cinammon 64bit / windows10
AMD A10-9700 RADEON R7, 10 compute cores 4c + 6g x 2
8GB RAM

**Comment by raystuttgart** [comment]
If restarting the game allows you to load the game again it is probably not an issue of the **mod**.
It is probably then rather an issue with the engine or that memory for some reason was full.

Honestly I really do not know what I should do if loading the save works perfectly fine after restart.
IT sounds like this would need to be fixed by the **developers of the game** and **not** by the **creators of a mod.**

**Comment by Nightinggale** [comment]
> Unhandled exception: page fault on read access to 0x0000241c in 32-bit code (0x01fff911).

0x0000241c is rather low so I will assume it's a NULL pointer with an offset. Since it works after a restart of the game, the most likely origin of the NULL pointer is if the game tried to allocate memory for a new object and failed, which it will if the biggest free block of memory is smaller than the memory being allocated. In other words running the game for a while has allocated and freed memory so often that the free memory has become too fragmented.

In other words it sounds like the game just ran into a memory limitation of being a 32 bit application. Nothing we can do about that. In fact the only thing, which can be done about it is moving to 64 bit, but we can't do that without the source code for the core game.

**Comment by RBRiddick** [comment]
Ok, clear. Sadly the problem is very annoying. I will try with a smaller map.
Thanks to both of you for the flash answers, and again, for your job.
greetings
